File: May-LeetCoding-Challenge/06-Majority-Element/Majority-Element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int val, n = 0;
        for (auto &x: nums){
            if (n == 0) {
                val = x;
                n++;
            }
            else n += (x != val) ? -1 : 1;
        }   
        return val;
    }
};

File: May-LeetCoding-Challenge/06-Majority-Element/Majority-Element.js
/**
 * @param {number[]} nums
 * @return {number}
 */

// Solution 1 sort and obtain the mid
var majorityElement = function(nums) {
    nums.sort(function(a,b){return a-b;});
    return nums[parseInt(nums.length/2)];
};
